openqa-qam-sle

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env bash
+
+# Search openQA jobs related to a given update inside job groups:
+# - Maintenance: Single Incidents / Core Incidents
+# - Maintenance: Aggregated updates / Core Maintenance Updates
+# Default search within the last 5 days for "Aggregated updates" builds
+# Usage: eg
+#   openqa-qam-sle -a SUSE:Maintenance:34576:337335
+#   openqa-qam-sle -a SUSE:Maintenance:34577:337331
+#   openqa-qam-sle -a SUSE:Maintenance:34564:337279
+#   openqa-qam-sle -a SUSE:Maintenance:34581:337626
+
+set -e
+
+# declare -a groups=(
+#     233 # 15-SP1
+#     306 # 15-SP2
+#     367 # 15-SP3
+#     439 # 15-SP4
+#     490 # 15-SP5
+#     546 # 15-SP6
+#     106 # 12-SP3
+#     282 # 12-SP5
+#     521 # 15-SP4-TERADATA
+#     191 # 12-SP3-TERADATA
+# )
+
+# Maintenance: Single Incidents / Core Incidents
+declare -A dict_group=()
+dict_group["15-SP1"]="233"
+dict_group["15-SP2"]="306"
+dict_group["15-SP3"]="367"
+dict_group["15-SP4"]="439"
+dict_group["15-SP5"]="490"
+dict_group["15-SP6"]="546"
+dict_group["12-SP3"]="106"
+dict_group["12-SP5"]="282"
+dict_group["15-SP4-TERADATA"]="521"
+dict_group["12-SP3-TERADATA"]="191"
+
+_ok() {
+    echo -e "\033[01;32m$1\033[0m"
+}
+
+_ko() {
+    echo -e "\033[01;31m$1\033[0m"
+}
+
+_warn() {
+    echo -e "\033[01;33m$1\033[0m"
+}
+
+_title() {
+    echo -e "\033[01;36m$1\033[0m"
+}
+
+search_maintenance_single_incidents() {
+    _title "Maintenance: Single Incidents / Core Incidents"
+    local ReviewRequestID=$1
+
+    local ID
+    ID="$(echo "$ReviewRequestID" | awk -F':' '{printf "%s", $3}')" # The ID of the update
+    # local RR="$(echo "$ReviewRequestID" | awk -F':' '{printf "%s", $4}')" # The Release Request of the update
+
+    local incident_settings
+    incident_settings=$(curl -sk -X GET "http://dashboard.qam.suse.de/api/incident_settings/$ID")
+
+    local build
+    build=$(echo "$incident_settings" | jq -r '.[0].settings.BUILD')
+
+    local versions
+    versions=$(echo "$incident_settings" | jq -r '.[].settings.VERSION' | sort | uniq)
+
+    local versions_teradata
+    versions_teradata=$(echo "$incident_settings" | jq -r '[.[]| { KEY:(.settings.VERSION + "|" +.settings.FLAVOR)}] | .[].KEY' | sort | uniq | grep TERADATA | sed -e 's/|.*-TERADATA$/-TERADATA/' | sort | uniq)
```

Comment:
Isn't it enough to run `sort | uniq` one time at the end?

Reply:
Could be, but I'll need to test it

Reply:
Removed duplicate filters in pipe